api/listener.go

```diff
@@ -109,6 +112,8 @@ func (cl *chainListener) AddResponder(responder apitypes.Responder) (string, err
 
 // RemoveResponder delete the responder
 func (cl *chainListener) RemoveResponder(listenerID string) (bool, error) {
+	cl.mu.Lock()
```

Comment:
TODO: line 126 & 127, if `cl.streamMap.Delete` may return false, we need to call it first before `r.Exit()`

Reply:
if the key is missing in the map, it has been returned in L119.